src/main/scala/beam/agentsim/infrastructure/parking/ParkingZoneFileUtils.scala

```diff
@@ -440,10 +478,28 @@ object ParkingZoneFileUtils extends ExponentialLazyLogging {
       val pricingModel = PricingModel(pricingModelString, newCostInDollarsString)
       val timeRestrictions = parseTimeRestrictions(timeRestrictionsString)
       val chargingPoint = ChargingPointType(chargingTypeString)
-      val numStalls = calculateNumStalls(numStallsString.toDouble)
+      val numStalls = calculateNumStalls(numStallsString.toDouble, reservedFor)
       val parkingZoneIdMaybe =
         if (parkingZoneIdString == null || parkingZoneIdString.isEmpty) None
         else Some(ParkingZone.createId(parkingZoneIdString))
+      val linkMaybe =
+        (if (locationXString == null || locationXString.isEmpty || locationYString == null || locationYString.isEmpty)
+           None
+         else
+           Some(new Coord(locationXString.toDouble, locationYString.toDouble))) match {
+          case Some(coord) if beamServices.isDefined =>
+            Some(
+              NetworkUtils.getNearestLink(beamServices.get.beamScenario.network, beamServices.get.geo.wgs2Utm(coord))
```

Comment:
We probably need to make this code GEO dependent. If GEO is Link then we can just use that link.

Reply:
How would you suggest to do that? because some of the parking stalls are link based but other not. In this scenario, mostly residential stalls are link, while rest are TAZ.

Reply:
In this case, this is the most appropriate way to do this.